src/third_party/catch/include/helpers.hpp

```diff
@@ -103,7 +103,10 @@
     auto collection_count_with_opts = libmongoc::collection_count_with_opts.create_instance();    \
     auto collection_create_index = libmongoc::collection_create_index.create_instance();          \
     auto collection_find = libmongoc::collection_find.create_instance();                          \
-    auto collection_aggregate = libmongoc::collection_aggregate.create_instance();
+    auto collection_aggregate = libmongoc::collection_aggregate.create_instance();                \
+    auto collection_get_name = libmongoc::collection_get_name.create_instance();                  \
+    collection_get_name->interpose([](mongoc_collection_t*){ return "dummy_collection"; });       \
+    auto collection_rename = libmongoc::collection_rename.create_instance();
```

Comment:
I see that you've added new mocked helpers to test this feature, did you have a chance to write some specific test cases for the operation?  I'd suggest interposing a method on rename() to test it mocked, and I think we should also add a test case for this to our integration tests, in `src/mongocxx/test/collection.cpp`.